numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,201 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def inplace_sum(items): 
+    """
+    Returns the sum of the elements of a list. 
+    Used in the follwing functions:
+    divergence,
+    """
+    it = iter(items)
+    total = next(it)
+    for item in it:
+        total += item
+    return total
+
+
+def divergence(v,*varargs):
+    """
+    Return the divergence of an N-dimensional vector field of N 
+    components, each of dimension N.
+
+    The divergence is computed using second order accurate central
+    differences in the interior and either first differences or second
+    order accurate one-sides (forward or backwards) differences at the
+    boundaries. The returned gradient hence has the same shape as the
+    input array.
+    
+    Parameters
+    ----------  
+    v : list of numpy arrays
+        Each of these array is the N-dimensional projection of the vector
+        field on the corresponding axis. 
+    
+    varargs : scalar or list of scalar, optional
+        N scalars specifying the sample distances for each dimension,
+        i.e. `dx`, `dy`, `dz`, ... Default distance: 1.
+        single scalar specifies sample distance for all dimensions.
+        if `axis` is given, the number of varargs must equal the number of axes.
+            
+    Returns
+    -------
+    divergence : numpy array
+        The output corresponds to the divergence of the input vector field. 
+        This means that the output array has the form 
+        dAx/dx + dAy/dy + dAz/dz + ... for an input vector field 
+        A = (Ax, Ay, Az, ...) up to N dimensions.
+    
+    Example
+    -------
+    This example shows how the calculated divergence of the 2-D field 
+    (0.5*x**2, -y*x) (whose divergence should be 0 everywhere) returns a 0 array
+    
+    >>> import numpy as np
+    >>> X,Y=np.mgrid[0:2000,0:2000]
+    >>> a1=0.5*X**2
+    >>> a2=-Y*X
+    >>> c=[a1,a2]
+    >>> d=np.divergence(c)
+    >>> print d
+    [[ 0.  0.  0. ...,  0.  0.  0.]
+     [ 0.  0.  0. ...,  0.  0.  0.]
+     [ 0.  0.  0. ...,  0.  0.  0.]
+     ..., 
+     [ 0.  0.  0. ...,  0.  0.  0.]
+     [ 0.  0.  0. ...,  0.  0.  0.]
+     [ 0.  0.  0. ...,  0.  0.  0.]
+    """
+    
+    N=len(v)
+    axes = tuple(range(N))
+    out = inplace_sum([gradient(v[ax],*varargs,axis=ax,edge_order=0)
```

Comment:
You'd have to pass the argument as an iterator here, otherwise the list with all the arrays still gets created. I think all you need to do is to remove the square brackets.

While you are at it, also follow PEP8 more closely: in particular, spaces around `=` (line 1388), spaces after `,`, and the `for ax in axes` should align with the first character after `inplace_sum(`
(If you use emacs, these kinds of checks can be done on-the-fly: http://docs.astropy.org/en/latest/development/codeguide_emacs.html)